internal/rbd/clone.go

```diff
@@ -197,41 +236,39 @@ func (rv *rbdVolume) createCloneFromImage(ctx context.Context, parentVol *rbdVol
 			}
 		}
 	}()
-	// flatten clone
-	errFlatten = tempClone.flattenRbdImage(ctx, rv.conn.Creds, false, rbdHardMaxCloneDepth, rbdSoftMaxCloneDepth)
-	if errFlatten != nil {
-		return errFlatten
-	}
-	// create snap of temp clone from temporary cloned image
-	// create final clone
-	// delete snap of temp clone
-	errClone = createRBDClone(ctx, tempClone, rv, cloneSnap, rv.conn.Creds)
-	if errClone != nil {
-		// set errFlatten error to cleanup temporary snapshot and temporary clone
-		errFlatten = errors.New("failed to create user requested cloned image")
-		return errClone
-	}
-	err = rv.getImageID()
-	if err != nil {
-		util.ErrorLog(ctx, "failed to get volume id %s: %v", rv, err)
-		return err
-	}
 
-	if parentVol.isEncrypted() {
-		err = parentVol.copyEncryptionConfig(&rv.rbdImage)
+	if rv.ThickProvision {
+		err = tempClone.DeepCopy(rv)
 		if err != nil {
-			return fmt.Errorf("failed to copy encryption config for %q: %w", rv, err)
+			return fmt.Errorf("failed to deep copy %q into %q: %w", parentVol, rv, err)
+		}
+	} else {
+		// flatten clone
+		errFlatten = tempClone.flattenRbdImage(ctx, rv.conn.Creds, false, rbdHardMaxCloneDepth, rbdSoftMaxCloneDepth)
+		if errFlatten != nil {
```

Comment:
@nixpanic if `Flatten` failed here, do we need to return error ? or can we move ahead without failing? I understand,  this is normal/general workflow we have so far/till now, so ideally should keep the same. But double confirming, is it mandatory to flatten before creating a snap from temporary clone image?.